projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3752,19 +3751,13 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             this.excelStyleFilteringComponent.overlayComponentId = id;
             return { id, ref: undefined };
         }
-        const ref = this.createComponentInstance(IgxGridExcelStyleFilteringComponent);
+        const ref = this.viewRef.createComponent(IgxGridExcelStyleFilteringComponent, { injector: this.injector } );
         ref.instance.initialize(column, this.overlayService);
         const id = this.overlayService.attach(ref.instance.element, options);
         ref.instance.overlayComponentId = id;
         return { ref, id };
     }
 
-    private createComponentInstance(component: any) {
-        const dynamicComponent: ComponentRef<any> = createComponent(component, { environmentInjector: this.envInjector, elementInjector: this.injector } );
-        this.appRef.attachView(dynamicComponent.hostView);
```

Comment:
Also this is the only thing using `appRef` in the grids, which leaves it unused, so you'll get `Property 'appRef' is declared but its value is never read` warnings and should remove it.

Reply:
Note: That will obv drag all other Grid (and Row Island) constructors. Not entirely sure if we should mark that as breaking for people extending the grid. At least this is targeting a major version.

Reply:
@kdinev should we treat removing the `appRef` from the constructors in any special way? May be add it as a Breaking change in the Changelog.md?

Reply:
@mtsvyatkova No, I don't think we should treat it in any special way or add it in the changelog